pkg/jwt/persistent/persistent.go

```diff
@@ -30,28 +29,52 @@ type TokenService struct {
 	jwks              json.RawMessage
 	gatewayClient     *client.Client
 	credOnlyGPTClient *gptscript.GPTScript
-	dispatcher        *dispatcher.Dispatcher
 	serverURL         string
 }
 
-func NewTokenService(ctx context.Context, serverURL string, gatewayClient *client.Client, dispatcher *dispatcher.Dispatcher, credOnlyGPTClient *gptscript.GPTScript) (*TokenService, error) {
-	key, err := ensureJWK(ctx, credOnlyGPTClient)
-	if err != nil {
-		return nil, err
-	}
-
+func NewTokenService(serverURL string, gatewayClient *client.Client, credOnlyGPTClient *gptscript.GPTScript) (*TokenService, error) {
 	t := &TokenService{
 		gatewayClient:     gatewayClient,
 		credOnlyGPTClient: credOnlyGPTClient,
-		dispatcher:        dispatcher,
 		serverURL:         serverURL,
 	}
+	return t, nil
+}
 
-	if err = t.replaceKey(ctx, key); err != nil {
-		return nil, err
+type TokenType string
+
+const (
+	TokenTypeRun TokenType = "run"
+)
+
+// EnsureJWK ensures that the JWK is created and stored in the GPTScript client. It should only be called in a controller pre-start hook which only allows one to be run at a time.
+func (t *TokenService) EnsureJWK(ctx context.Context) error {
+	_, err := ensureJWK(ctx, t.credOnlyGPTClient)
```

Comment:
nit: I believe this is the only place `ensureJWK` is used and we don't need the first returned parameter. It can be removed.